### PR TITLE
Add a no-short-mac FIPS indicator

### PR DESCRIPTION
**Pull request by paulidale**
KMAC is not permitted to produce less than 32 bits of output as per SP 800-185 8.4.2.  This change enforces this.

HMAC is also subject to the same limitation but our implementation only outputs the digest block size.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: include/openssl/fips_names.h
Comment (slontis) [inline include/openssl/fips_names.h:67]: enabled??

File: providers/common/securitycheck_default.c
Comment (slontis) [inline providers/common/securitycheck_default.c:32]: just use the FIPS_ function and get rid of this. It has no purpose now that we have to do #ifdef FIPS_MODULE where we use it.

File: include/openssl/fips_names.h
Comment (paulidale) [reply include/openssl/fips_names.h:67]: That's what **all** the other Booleans in the file say.

File: providers/common/securitycheck_default.c
Comment (slontis) [reply providers/common/securitycheck_default.c:32]: At some point after everything is merged from lots of PR's we should probably change the FIPS_ name to an ossl_ name

File: providers/fips/fipsprov.c
Comment (slontis) [inline providers/fips/fipsprov.c:144]: This is one of the points where merging may go really bad with all the PR's

Comment (paulidale) [reply providers/fips/fipsprov.c:144]: That's why I bumped it by two -- there is a second PR that wants to modify this.
Maybe set it to something large with an assert if it overflows.

File: test/recipes/30-test_evp_data/evpmac_common.txt
Comment (slontis) [inline test/recipes/30-test_evp_data/evpmac_common.txt:545]: evp_test.c needs to handle this parameter

Comment (paulidale) [reply test/recipes/30-test_evp_data/evpmac_common.txt:545]: Done

**Comment by slontis** [comment]
Are the test failures valid?

File: doc/man7/EVP_MAC-KMAC.pod
Comment (slontis) [inline doc/man7/EVP_MAC-KMAC.pod:71]: should we be more specific here about what the restriction actually is?

**Comment by paulidale** [comment]
Still digging into the test failures.  They appear to be in CMS and are failing on master.  At least mostly.

File: doc/man7/EVP_MAC-KMAC.pod
Comment (paulidale) [reply doc/man7/EVP_MAC-KMAC.pod:71]: Added

File: doc/man7/EVP_MAC-KMAC.pod
Comment (slontis) [inline doc/man7/EVP_MAC-KMAC.pod:154]: Is the format ````[SP800..]https://...````

Comment (paulidale) [reply doc/man7/EVP_MAC-KMAC.pod:154]: It displays as `SP 800-185 8.4.2` and the link works, so the format is okay.

**Review by slontis — APPROVED** [review]

**Comment by slontis** [comment]
Test failures are not caused by this PR.

**Comment by slontis** [comment]
When merging any of these indicator PR's we need to very careful with anywhere where we are changing  fixed arrays sizes. Multiple PR's are all hitting these same arrays - so there WILL be merge issues where the size needs to increase,

**Comment by slontis** [comment]
(I know changing to OSSL_PARAM_BUILD might fix this, but it is too late to do that change now - and seems like overkill)

**Comment by paulidale** [comment]
ping for review @openssl/committers

File: apps/fipsinstall.c
Comment (tom-cosgrove-arm) [inline apps/fipsinstall.c:67]: There's a slight inconsistency between `"no_short_mac"` and `OPT_NO_SHORT_MACS`

**Review by tom-cosgrove-arm — APPROVED** [review]
With other parameters we've sometimes put something in `CHANGES.md` - did you want to do that with this one?

Either way, approved

**Comment by paulidale** [comment]
Added a changes entry and resolved the naming difference.
Reconfirmation please.
Don't see a need for a timer reset.

File: CHANGES.md
Comment (tom-cosgrove-arm) [inline CHANGES.md:36]: Shane or Paul?

**Review by tom-cosgrove-arm — APPROVED** [review]
Reconfirming, although not sure about the attribution in `CHANGES.md` - everything else looks good

**Comment by paulidale** [comment]
Attribution fixed and rebased to remove conflicts.

**Comment by tom-cosgrove-arm** [comment]
Hmm, CI failed at `Provider compatibility for PRs / development-branches (PR, PR.tar.gz)`

**Review by slontis — APPROVED** [review]
I would prefer to just do a second pass for the CHANGES.md after we have completed all the FIPS indicator work. It is just another point for merge conflicts.

**Comment by tom-cosgrove-arm** [comment]
@slanterns Paul has fixed `CHANGES.md` - I would re-confirm but for the failing CI - is this expected? If so, consider this a reconfirmation

**Comment by slanterns** [comment]
> @slanterns Paul has fixed `CHANGES.md` - I would re-confirm but for the failing CI - is this expected? If so, consider this a reconfirmation

Hmm, I have no idea on why do you mention me?

**Comment by paulidale** [comment]
The failing CI was a result of a different problem which has been merged now.
This results in some versioning problems appearing.

**Comment by paulidale** [comment]
Got the tests passing.  Reapprovals required (again).

File: CHANGES.md
Comment (slontis) [inline CHANGES.md:1117]: The option uses _ and  the config file uses - (hence the confusion) :)

**Review by slontis — APPROVED** [review]
I probably would have put those test changes in another PR if they have nothing to do with this PR..

**Review by tom-cosgrove-arm — APPROVED** [review]
Reconfirming.

I too would have been happy with "CI failure is unrelated" and a separate PR, but this is okay too

**Comment by paulidale** [comment]
Merged resolving trivial conflicts.
Didn't reset timer for the minor test case changes.

Thanks for the feedback.

**Pull request closed without merging**
